### PR TITLE
feat: add ability to change panel type

### LINKED ISSUE #709: Ability to change type of a panel
## Is your feature request related to a problem? No

It would be quite useful to convert a panel to/from time-series -> numeric and back. Datadog has this ability in the Edit of a panel and this would be a compelling/competing feature. 

## Describe the solution you'd like

In the edit options add a type drop-down which will convert the panel to the other type (and in future -> to more possible types). Since there are only two types now this could be a toggle button as well.

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #709 

## Demo
https://www.loom.com/share/ef00628578564526a31a417343e2df70

### CODE REVIEW

File: frontend/src/container/NewWidget/index.tsx [hunk 1]
```diff
@@ -149,11 +153,18 @@ function NewWidget({
...
 		selectedTime.enum,
 		selectedWidget?.id,
-		selectedGraph,
 		getQueryResults,
 		globalSelectedInterval,
+		graphType,
 	]);
 
+	const setGraphHandler = (type: ITEMS): void => {
+		const url = new URL(window.location.href);
```
Comment (palashgdev) [inline frontend/src/container/NewWidget/index.tsx:162]: can we use useLocation hook here

File: frontend/src/container/NewWidget/index.tsx [hunk 2]
```diff
@@ -149,11 +153,18 @@ function NewWidget({
...
+		url.searchParams.set('graphType', type);
+		window.history.replaceState({}, '', url);
```
Comment (palashgdev) [inline frontend/src/container/NewWidget/index.tsx:164]: can we use history from `lib/history`

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/NewWidget/RightContainer/index.tsx [hunk 3]
```diff
@@ -34,14 +38,15 @@ function RightContainer({
...
-				disabled
-			/>
-
+			<Title>Panel Type</Title>
+			<Select
+				onChange={(value: ITEMS): void => setGraphHandler(value)}
+				value={selectedGraph}
+				style={{ width: '100%' }}
+			>
+				<Option value="VALUE">Value</Option>
```
Comment (palashgdev) [inline frontend/src/container/NewWidget/RightContainer/index.tsx:47]: can we make use of this config `/frontend/src/container/NewDashboard/ComponentsSlider/menuItems.ts`

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Comment by gitstart** [comment]
Hi @palashgdev, requested changes have been made. Please take a look 🙏

**Review by palashgdev — APPROVED** [review]

**Comment by srikanthccv** [comment]
Just by looking at the change, I don't see anywhere where the underlying query was modified, which is the real functional part of the change panel type. Is this tested?

**Comment by palashgdev** [comment]
> Just by looking at the change, I don't see anywhere where the underlying query was modified, which is the real functional part of the change panel type. Is this tested?

modification of the query is not incorporated for more info I am disabling this dropdown
https://github.com/SigNoz/signoz/issues/2410

but from this PR at least we have a way to listen onChange

**Pull request merged**